lib/dealership.rb

```diff
@@ -0,0 +1,64 @@
+class Dealership
+  attr_reader :name, :address, :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+    @total_value = 0
+
+  end
+
+  def inventory_count
+    @inventory.count
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    if @inventory.length == 0
+      false
+    else
+      true
+    end
+  end
+
+  def cars_by_make(make)
+    car_makes = []
+    @inventory.each do |car|
+      if car.make == make
+        car_makes << car
+      end
+    end
+    car_makes
+  end
+
+  def total_value
+    @tot_value = 0
+    @inventory.each do |car|
+      @tot_value += car.total_cost
+    end
+    return @tot_value
+  end
```

Comment:
Rather than using an instance variable `@tot_value` I would use a local variable to do the calculation

```suggestion
  def total_value
    tot_value = 0
    @inventory.each do |car|
      tot_value += car.total_cost
    end
    return tot_value
  end
```

Then everywhere in the code you would call `total_value`. 

If you do it this way and you try to call `average_price_of_car` or `details` on dealership before you every call `total_value` you will get @tot_value == nil